pyccel/ast/basic.py

```diff
@@ -178,6 +261,9 @@ def substitute(self, original, replacement, excluded_nodes = ()):
                       The object which will be inserted instead
         excluded_nodes : tuple of types
                       Types for which substitute should not be called
+        invalidate : bool
+                    Indicates whether the removed object should
+                    be invalidated
```

Comment:
`invalidate` means that the `invalidate_node` method is called. As per the docstring of `invalidate_node`:

> Indicate that this node is temporary.
> This will allow it to remove itself from its children's users.
> If a child subsequently has no users, invalidate_node is called recursively

Reply:
And with `child`/`children` you mean `attribute_nodes`?

Reply:
Yes. I will update the docstring

Reply:
Fixed 458adf9cdb8a3a2e519ce33cff71234e1a806e00